equinix/resource_metal_organization_member.go

```diff
@@ -0,0 +1,275 @@
+			},
+			"created": {
+				Type:        schema.TypeString,
+				Description: "When the invitation was created (only known in the invitation stage)",
+				Computed:    true,
+			},
+			"updated": {
+				Type:        schema.TypeString,
+				Description: "When the invitation was updated (only known in the invitation stage)",
+				Computed:    true,
+			},
+			"roles": {
+				Type:        schema.TypeSet,
+				Description: "Organization roles",
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				// TODO: Update should be supported. packngo.InvitationService does not offer an Update	method.
+				ForceNew: true,
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Description: "The state of the membership ('invited' when an invitation is open, 'active' when the user is an organization member)",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceMetalOrganizationMemberCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).metal
+
+	email := d.Get("invitee").(string)
+	createRequest := &packngo.InvitationCreateRequest{
+		Invitee:     email,
+		Roles:       convertStringArr(d.Get("roles").(*schema.Set).List()),
+		ProjectsIDs: convertStringArr(d.Get("projects_ids").(*schema.Set).List()),
+		Message:     strings.TrimSpace(d.Get("message").(string)),
+	}
+
+	orgID := d.Get("organization_id").(string)
+	_, _, err := client.Invitations.Create(orgID, createRequest, nil)
+	if err != nil {
+		return friendlyError(err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", email, orgID))
+
+	return resourceMetalOrganizationMemberRead(d, meta)
+}
+
+func findMember(invitee string, members []packngo.Member, invitations []packngo.Invitation) (*member, error) {
+	for _, mbr := range members {
+		if mbr.User.Email == invitee {
+			return &member{Member: &mbr}, nil
+		}
+	}
+
+	for _, inv := range invitations {
+		if inv.Invitee == invitee {
+			return &member{Invitation: &inv}, nil
+		}
+	}
+	return nil, fmt.Errorf("member not found")
+}
+
+func resourceMetalOrganizationMemberRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).metal
+	parts := strings.Split(d.Id(), ":")
+	invitee := parts[0]
+	orgID := parts[1]
+
+	listOpts := &packngo.ListOptions{Includes: []string{"user"}}
+	invitations, _, err := client.Invitations.List(orgID, listOpts)
+	if err != nil {
+		err = friendlyError(err)
+		// If the org was destroyed, mark as gone.
+		if isNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	members, _, err := client.Members.List(orgID, &packngo.GetOptions{Includes: []string{"user"}})
+	if err != nil {
+		err = friendlyError(err)
+		// If the org was destroyed, mark as gone.
+		if isNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	member, err := findMember(invitee, members, invitations)
+	if !d.IsNewResource() && err != nil {
+		return friendlyError(fmt.Errorf("Could not find member %s in organization %s", invitee, orgID))
```

Comment:
If member leave the project we don't want an error but a WARN and remove the id from the state

Reply:
fixed in 98a070f